Xamarin.PropertyEditing.Mac/Controls/EnumEditorControl.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections;
+using System.ComponentModel;
+using System.Diagnostics;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class EnumEditorControl : PropertyEditorControl
+	{
+		NSComboBox ComboBoxEditor;
+
+		dynamic EnumEditorViewModel => ViewModel;
```

Comment:
Also my generic fu may not be too hot but AFACT you can cast the generic CreateInstance as 
`(PropertyViewModel) Activator.CreateInstance (type, property, this.editors);`
but you couldn't do 
`var epvm = (EnumPropertyViewModel<>)Activator.CreateInstance (type, property, this.editors);`
or 
`var epvm = (EnumPropertyViewModel<type>)Activator.CreateInstance (type, property, this.editors);`
Which I think would mean that to access the ValueName property we'd have to jump through a  few hoops by calling...
`var epvm = (PropertyViewModel)Activator.CreateInstance (type, property, this.editors);
var epvmType = epvm.GetType();
var valueNameProperty = epvmType.GetProperty("ValueName");
var valueNameValue = (string)valueNameProperty.GetValue (epvm);`

Am I missing a simpler way of doing this? That seems quite convoluted when we can leverage the power of the DLR and just call 
`EnumEditorViewModel.ValueName`
rather than the forementioned hoops.
Running the code here doesn't seem that slow. Its not like we are on i386 processors these days :)